### ISSUE TITLE
Resource tree

### ISSUE THREAD

**Issue by tzachshabtay** [tzachshabtay/MonoAGS#290]
> (Crimson Wizard): @tzachs, BTW, do you think it may make sense to provide a "file tree" interface to iterate asset entries, instead of call "load" for them? (Not sure if it actually always loads for real though).
In other words, instead of (or along with) IResourceLoader, have something like IResourceTree with dictionary of all assets and possibility to request their loading.

> For instance, in AGS the engine has a full list of all assets in the package (not loaded to memory, just a list of their names and some info). It's not available in script API, but could be exposed there also in theory.

> Hmm, I'd imagine this would need "Rescan" function, and listen for disk changes, because files on disk may get modified. Well, ones in package may change also, if its something like zip archive.

> (Monsieur OUXX): I actually just did that with my AssetDefsFactory.Process(wildcard). Internally it's not a tree but that's the idea.

**Comment by tzachs** [comment]
@tzachshabtay I think you got the wrong tzach

**Comment by ivan-mogilko** [comment]
@tzachs , lol, that text is copy/pasted from the AGS forums, where we have only one tzachs, and when copied here with @ sign it links to another github profile :)

**Comment by jeancallisti** [comment]
If you want adopters you should mimic the AGS interface, or, even better, modern assets concepts such as the ones in Unity. So it means yes there should be a basic form of resources browser, in a way or another.

**Comment by ivan-mogilko** [comment]
@jeancallisti IIRC this was about program interface (in code/script), not graphic interface.

**Comment by tzachshabtay** [comment]
Right, this issue is about a programmable API for iterating over the resources. There will definitely be some form of "asset/resource browser" in the editor.
